tests/test_jexl.py

```diff
@@ -118,6 +121,34 @@ def test_validate():
     assert errors == ['Could not parse expression: "\n"']
 
 
+JEXL_ALPHABET = strategies.characters(whitelist_categories=(
+    # Letters
+    'Lu', 'Ll',
+    # Numbers
+    'Nd', 'No',
+    # Spaces
+    'Zs', 'Zl', 'Zp',
+), whitelist_characters=[
+    '\t', '\n', '\r', ' ',
+    '$', '|', '+', '/',
+    '(', ')', '{', '}', '[', ']',
+    "'", '"', ',', ':', '?',
+])
+
+
+@hypothesis.given(strategies.text(JEXL_ALPHABET))
+@hypothesis.settings(max_examples=500)
+def test_validate_always_returns(s):
+    jexl = JEXL()
+    errors = list(jexl.validate(s))
+    assert isinstance(errors, list)
+
+    # I couldn't figure out how to generate '{{' and '}}' (or, indeed,
```

Comment:
This is a strange comment. It looks like it belongs in a GitHub PR rather than the code. 
Some people use the notation 
```
FIXME(peterbe): Couldn't figure out a good way to ...
```

Reply:
As for the comment itself, it looks like the text strategy [builds on a StringStrategy class](https://github.com/HypothesisWorks/hypothesis-python/blob/master/src/hypothesis/strategies.py#L858) that could be used to build a custom token-based strategy instead of a character-based one. I'm not sure how worth it it is to do so, though.